docs/explanation/overlays.rst

```diff
@@ -0,0 +1,185 @@
+****************
+``OVERLAY`` Step
+****************
+
+It is sometimes necessary to modify the base filesystem used in when generating
```

Comment:
> the base filesystem used in when generating

this construct sounds strange to me.

I would also add a ref link to base (if it exists).

Reply:
Yeah that's because it's just all kinds of awkward :-) I've changed it to "modify the base filesystem when generating a parts-based payload," though I'm open to other suggestions for better ways to introduce overlays.

Unfortunately there's no base ref in the craft-parts doc yet and I couldn't see anything in other projects that would be relevant.

Reply:
maybe we should explicitly call-out rockcraft here? Something like "Some Craft applications, like Rockcraft, include entire base filesystems in addition to the usual part-generated payload. For those applications, it is often desirable to modify this base filesystem..."
that's not _quite_ right, but something along those paths?

Reply:
I like that — it anchors the reason for the overlay step to a specific real-world use case so the reader can more quickly decide whether the overlay step is relevant to them.